### PR TITLE
#16 pirate rolls

### PR DESCRIPTION
**Pull request by pickardj79**
The main change is for pirates cards "penalty" to be marking players who lost to the pirates in the database as having lost (column `card_action_choice`). Then, when the pirates have been defeated or all players have faced them, the hazards are applied to players who lost. A new card method `finishCard` allows pirates to shift to the penalty phase and all other cards to draw the next card.

I was also able to combine the code for applying hazards from both meteors and pirates. The meteors is done similarly to how the pirates was accomplished - by setting the `card_action_choice` of all players still in flight to "lost".

### CODE REVIEW

File: modules/GT_Enemy.php [hunk 1]
```diff
@@ -83,12 +83,10 @@ function applyPenalty()
       ['player_name' => $this->player['player_name'], 'type' => $this->card->getType()]
     );
 
-    $game = $this->game;
-
     if ($this->card instanceof HazardCard) {
-      return $this->card->applyPenalty($game, $this->player);
+      return $this->card->applyPenalty($ths->game, $this->player);
```
Comment (demmerichs) [inline modules/GT_Enemy.php]: typo

**Review by demmerichs — COMMENTED** [review]
I looked over the changes and got the gist, but did not try too hard to understand all the details ;) I tested the branch, and the meteors and pirates worked as expected now!
Just found one typo :)

File: modules/GT_Enemy.php (hunk 1 above)
Comment (demmerichs) [reply modules/GT_Enemy.php]: and I just realized that this typo was removed in a follow-up commit

Comment (pickardj79) [reply modules/GT_Enemy.php]: ha - nice. thanks for looking it over

**Review by kuwizard — APPROVED** [review]

**Comment by kuwizard** [comment]
Great job @pickardj79!

**Pull request merged**